### ISSUE TITLE
Crash with -ast-dump=json

### ISSUE THREAD

**Issue by kleag** [llvm/llvm-project#57651]
The following command crashes. The same with no `=json` works. This could be related to #48838 but I'm not sure.

```shell
clang -x c++ -std=c++17 -fPIC -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I/home/gael/Projets/Lima/lima/lima_common/src/ -I/home/gael/Projets/Lima/lima/lima_common/src/common/XMLConfigurationFiles -Xclang -ast-dump -fsyntax-only test/xmlConfigurationFileHandler.h
```

The source file `xmlConfigurationFileHandler.h` can be found [here](https://github.com/aymara/lima/blob/master/lima_common/src/common/XMLConfigurationFiles/xmlConfigurationFileParser.h).

```console
                            "end": {
                              "offset": 110497,
                              "col": 80,
                              "tokLen": 1,
                              "includedFrom": {
                                "file": "/usr/include/x86_64-linux-gnu/qt5/QtCore/QString"
                              }
                            }
                          },
                          "isReferenced": true,
                          "name": "argBases"Stack dump:
0.      Program arguments: clang -x c++ -std=c++17 -fPIC -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I/home/gael/Projets/Lima/lima/lima_common/src/ -I/home/gael/Projets/Lima/lima/lima_common/src/common/XMLConfigurationFiles -Xclang -ast-dump=json -fsyntax-only test/xmlConfigurationFileHandler.h 
1.      <eof> parser at end of file
2.      /usr/include/x86_64-linux-gnu/qt5/QtCore/qstring.h:2114:20: Mangling declaration 'argBases'
 #0 0x00007f05473d74ff llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/usr/lib/x86_64-linux-gnu/libLLVM-10.so.1+0x9814ff)
 #1 0x00007f05473d57b0 llvm::sys::RunSignalHandlers() (/usr/lib/x86_64-linux-gnu/libLLVM-10.so.1+0x97f7b0)
 #2 0x00007f05473d6c4d llvm::sys::CleanupOnSignal(unsigned long) (/usr/lib/x86_64-linux-gnu/libLLVM-10.so.1+0x980c4d)
 #3 0x00007f054732ce60 (/usr/lib/x86_64-linux-gnu/libLLVM-10.so.1+0x8d6e60)
 #4 0x00007f054dba6420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #5 0x00007f054bc557b5 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb6b7b5)
 #6 0x00007f054bc520a1 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb680a1)
 #7 0x00007f054bc50620 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb66620)
 #8 0x00007f054bc4ceb1 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb62eb1)
 #9 0x00007f054bc78a90 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb8ea90)
#10 0x00007f054bc7806a (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb8e06a)
#11 0x00007f054bc7817e clang::ASTNameGenerator::getName[abi:cxx11](clang::Decl const*) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb8e17e)
#12 0x00007f054bc6c89f clang::JSONNodeDumper::VisitNamedDecl(clang::NamedDecl const*) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb8289f)
#13 0x00007f054bc6d076 clang::JSONNodeDumper::VisitVarDecl(clang::VarDecl const*) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb83076)
#14 0x00007f054bc63d52 clang::JSONNodeDumper::Visit(clang::Decl const*) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0xb79d52)
#15 0x00007f054ba62bc5 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x978bc5)
#16 0x00007f054ba6f67d (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x98567d)
#17 0x00007f054ba69b49 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x97fb49)
#18 0x00007f054ba64ed4 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x97aed4)
#19 0x00007f054ba651d7 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x97b1d7)
#20 0x00007f054ba69b0d (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x97fb0d)
#21 0x00007f054ba6e17c (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x98417c)
#22 0x00007f054ba62c33 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x978c33)
#23 0x00007f054ba6f67d (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x98567d)
#24 0x00007f054ba629c4 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x9789c4)
#25 0x00007f054ba6ba47 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x981a47)
#26 0x00007f054ba6b8e5 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x9818e5)
#27 0x00007f054ba62bda (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x978bda)
#28 0x00007f054ba6f67d (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x98567d)
#29 0x00007f054ba629c4 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x9789c4)
#30 0x00007f054ba62d03 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x978d03)
#31 0x00007f054ba6f67d (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x98567d)
#32 0x00007f054ba629c4 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x9789c4)
#33 0x00007f054ba62d03 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x978d03)
#34 0x00007f054ba6278e (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x97878e)
#35 0x00007f054ba5765e clang::Decl::dump(llvm::raw_ostream&, bool, clang::ASTDumpOutputFormat) const (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x96d65e)
#36 0x00007f054b93fc13 clang::ParseAST(clang::Sema&, bool, bool) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x855c13)
#37 0x00007f054cdb5e58 clang::FrontendAction::Execute() (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x1ccbe58)
#38 0x00007f054cd6e8a1 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x1c848a1)
#39 0x00007f054ce19daf clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x1d2fdaf)
#40 0x000000000041229d cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/lib/llvm-10/bin/clang+0x41229d)
#41 0x00000000004105b1 (/usr/lib/llvm-10/bin/clang+0x4105b1)
#42 0x00007f054cabf8f2 (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x19d58f2)
#43 0x00007f054732cc67 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/usr/lib/x86_64-linux-gnu/libLLVM-10.so.1+0x8d6c67)
#44 0x00007f054cabee2f clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, bool*) const (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x19d4e2f)
#45 0x00007f054ca9752f clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x19ad52f)
#46 0x00007f054ca976da clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) const (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x19ad6da)
#47 0x00007f054caaa93c clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) (/usr/lib/x86_64-linux-gnu/libclang-cpp.so.10+0x19c093c)
#48 0x000000000041002f main (/usr/lib/llvm-10/bin/clang+0x41002f)
#49 0x00007f0546502083 __libc_start_main /build/glibc-SzIz7B/glibc-2.31/csu/../csu/libc-start.c:342:3
#50 0x000000000040d7ce _start (/usr/lib/llvm-10/bin/clang+0x40d7ce)
clang: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 10.0.0-4ubuntu1 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
clang: error: unable to execute command: Segmentation fault
clang: note: diagnostic msg: Error generating preprocessed source(s).
```

clang comes from KDE neon packages:
```console
$ clang --version
clang version 10.0.0-4ubuntu1 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by kleag** [comment]
I just tried with clang-12 from packages. The same crash occurred.

**Comment by EugeneZelenko** [comment]
Could you please try `main` branch? https://godbolt.org should be helpful.

**Comment by kleag** [comment]
I did not know this tool. Very useful. I will try with it.
I just tried with clang 14 available in [this docker image](https://hub.docker.com/repository/docker/aymara/manylinux_2_24_with_llvm14.0.6) and it worked on my example file. Both in the container and after retrieving it on the KDE Neon host. 
I'm also compiling the main branch on a machine that should have enough space available.

As it works with the clang 14 I tried, should we close this issue?
